### PR TITLE
BUG: fix alignment in series ops (GH14227)

### LINKED ISSUE #14227: Labels mismatch when adding series with repeated index values in 0.19rc1
This showed up in the dask.dataframe test suite when testing against the 0.19 release candidate.  I'm unsure if this was an intended change or not:

### Pandas 0.18

```python
In [1]: import pandas as pd

In [2]: a = pd.Series([1, 2], index=[1, 1])

In [3]: b = pd.Series([10, 10], index=[1, 2])

In [4]: a
Out[4]: 
1    1
1    2
dtype: int64

In [5]: b
Out[5]: 
1    10
2    10
dtype: int64

In [6]: a + b
Out[6]: 
1    11.0
1    12.0
2     NaN
dtype: float64
```

### Pandas 0.19rc1

```python
In [1]: import pandas as pd

In [2]: a = pd.Series([1, 2], index=[1, 1])

In [3]: b = pd.Series([10, 10], index=[1, 2])

In [4]: a
Out[4]: 
1    1
1    2
dtype: int64

In [5]: b
Out[5]: 
1    10
2    10
dtype: int64

In [6]: a + b
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-6-f96fb8f649b6> in <module>()
----> 1 a + b

/home/mrocklin/Software/anaconda/lib/python3.5/site-packages/pandas/core/ops.py in wrapper(left, right, name, na_op)
    671             return NotImplemented
    672 
--> 673         left, right = _align_method_SERIES(left, right)
    674 
    675         converted = _Op.get_op(left, right, name, na_op)

/home/mrocklin/Software/anaconda/lib/python3.5/site-packages/pandas/core/ops.py in _align_method_SERIES(left, right, align_asobject)
    613                                                 return_indexers=True)
    614             # if DatetimeIndex have different tz, convert to UTC
--> 615             left.index = index
    616             right.index = index
    617 

/home/mrocklin/Software/anaconda/lib/python3.5/site-packages/pandas/core/generic.py in __setattr__(self, name, value)
   2754         try:
   2755             object.__getattribute__(self, name)
-> 2756             return object.__setattr__(self, name, value)
   2757         except AttributeError:
   2758             pass

pandas/src/properties.pyx in pandas.lib.AxisProperty.__set__ (pandas/lib.c:44336)()

/home/mrocklin/Software/anaconda/lib/python3.5/site-packages/pandas/core/series.py in _set_axis(self, axis, labels, fastpath)
    306         object.__setattr__(self, '_index', labels)
    307         if not fastpath:
--> 308             self._data.set_axis(axis, labels)
    309 
    310     def _set_subtyp(self, is_all_dates):

/home/mrocklin/Software/anaconda/lib/python3.5/site-packages/pandas/core/internals.py in set_axis(self, axis, new_labels)
   2774             raise ValueError('Length mismatch: Expected axis has %d elements, '
   2775                              'new values have %d elements' %
-> 2776                              (old_len, new_len))
   2777 
   2778         self.axes[axis] = new_labels

ValueError: Length mismatch: Expected axis has 3 elements, new values have 5 elements
```

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
- [x] closes #14227
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/ops.py [hunk 1]
```diff
@@ -609,11 +609,14 @@ def _align_method_SERIES(left, right, align_asobject=False):
 
             left, right = left.align(right, copy=False)
 
-            index, lidx, ridx = left.index.join(right.index, how='outer',
-                                                return_indexers=True)
-            # if DatetimeIndex have different tz, convert to UTC
-            left.index = index
-            right.index = index
+            if is_datetime64tz_dtype(left.index):
```
Comment (jreback) [inline pandas/core/ops.py]: should be inside ``.join`` not here

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/ops.py (hunk 1 above)
Comment (jorisvandenbossche) [inline pandas/core/ops.py]: Do you mean in `align` ?

Comment (jreback) [inline pandas/core/ops.py]: yes, that's what I mean

**Comment by jreback** [comment]
@jorisvandenbossche can you update

**Comment by jorisvandenbossche** [comment]
@jreback I moved this to align (and therefore also fixing the bug in align), but it is still not yet that clean ... 
So the problem for tz aware indexes is that `Index.join` does not returns indexers when the result does not need to be reindexed (although in case of tz aware indexes the actual indexes can be different and have to be adapted), and based on that the index of the result is adapted in `align` or not. So I still have to check for this specific case. I tried to solve it in `Index.join` itself, but that caused other errors.

**Comment by jreback** [comment]
if this works (and you want to push in 0.19.0), ok.. maybe just add a TODO / issue to fix/clean later.

**Comment by jorisvandenbossche** [comment]
Well, it was a hack before, and it is still the exact same hack now, only moved down the stack into align (fixing the behaviour of align as well), so merging for now.

**Pull request merged**